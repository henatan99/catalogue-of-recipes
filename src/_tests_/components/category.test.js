import React from 'react';
import { create } from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom'

import Category from '../../components/category';

test('renders when function prop is set to null', () => {
  
  const props = {
    name: 'name',
    image: 'image',
    recipes: 0,
    handleFilterChange: null,
  }

  const jsx = (
    <BrowserRouter>
      <Category {...props} />
    </BrowserRouter>
  )

  const tree = create(jsx).toJSON();
  expect(tree).toMatchSnapshot();

});

test('renders when function prop is set to mock function', () => {
  const handleClick = jest.fn();
  const props = {
    name: 'name',
    image: 'image',
    recipes: 0,
    handleFilterChange: handleClick,
  }

  const jsx = (
    <BrowserRouter>
      <Category {...props} />
    </BrowserRouter>
  )

  const tree = create(jsx).toJSON();
  expect(tree).toMatchSnapshot();

})
