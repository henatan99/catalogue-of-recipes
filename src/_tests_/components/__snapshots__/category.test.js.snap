// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when function prop is set to mock function 1`] = `
<a
  className="card"
  href="/Meals/name"
  onClick={[Function]}
>
  <img
    alt="category-img"
    className="category-img card-img"
    src="image"
  />
  <div
    className="card-info"
  >
    <h2
      className="recipe card-title"
    >
      name
    </h2>
    <span
      className="recipe-num card-span"
    >
      0
       
      recipes
    </span>
  </div>
</a>
`;

exports[`renders when function prop is set to null 1`] = `
<a
  className="card"
  href="/Meals/name"
  onClick={[Function]}
>
  <img
    alt="category-img"
    className="category-img card-img"
    src="image"
  />
  <div
    className="card-info"
  >
    <h2
      className="recipe card-title"
    >
      name
    </h2>
    <span
      className="recipe-num card-span"
    >
      0
       
      recipes
    </span>
  </div>
</a>
`;

exports[`renders when function prop is set to null 2`] = `
<a
  className="card"
  href="/Meals/name"
  onClick={[Function]}
>
  <img
    alt="category-img"
    className="category-img card-img"
    src="image"
  />
  <div
    className="card-info"
  >
    <h2
      className="recipe card-title"
    >
      name
    </h2>
    <span
      className="recipe-num card-span"
    >
      0
       
      recipes
    </span>
  </div>
</a>
`;

exports[`renders when mock function prop given 1`] = `
<a
  className="card"
  href="/Meals/name"
  onClick={[Function]}
>
  <img
    alt="category-img"
    className="category-img card-img"
    src="image"
  />
  <div
    className="card-info"
  >
    <h2
      className="recipe card-title"
    >
      name
    </h2>
    <span
      className="recipe-num card-span"
    >
      0
       
      recipes
    </span>
  </div>
</a>
`;

exports[`renders with no prop 1`] = `
<a
  className="card"
  href="/Meals/name"
  onClick={[Function]}
>
  <img
    alt="category-img"
    className="category-img card-img"
    src="image"
  />
  <div
    className="card-info"
  >
    <h2
      className="recipe card-title"
    >
      name
    </h2>
    <span
      className="recipe-num card-span"
    >
      0
       
      recipes
    </span>
  </div>
</a>
`;

exports[`some description 1`] = `
<a
  className="card"
  href="/Meals/name"
  onClick={[Function]}
>
  <img
    alt="category-img"
    className="category-img card-img"
    src="image"
  />
  <div
    className="card-info"
  >
    <h2
      className="recipe card-title"
    >
      name
    </h2>
    <span
      className="recipe-num card-span"
    >
      0
       
      recipes
    </span>
  </div>
</a>
`;
